src/state/sagas/scenario/UpdateAndLaunchScenario/UpdateAndLaunchScenario.js

```diff
@@ -1,34 +1,63 @@
 // Copyright (c) Cosmo Tech.
 // Licensed under the MIT license.
 
-import { takeEvery, call } from 'redux-saga/effects';
+import { takeEvery, call, put } from 'redux-saga/effects';
 import { SCENARIO_ACTIONS_KEY } from '../../../commons/ScenarioConstants';
 import { ORGANISATION_ID } from '../../../../configs/App.config';
 import ScenarioService from '../../../../services/scenario/ScenarioService';
-// import { STATUSES } from '../../../commons/Constants';
+import ScenariorunService from '../../../../services/scenariorun/ScenariorunService';
+import { STATUSES } from '../../../commons/Constants';
 
 // generators function
 export function * updateAndLaunchScenario (action) {
-  // yield keyword is here to milestone and save the action
   const workspaceId = action.workspaceId;
   const scenarioId = action.scenarioId;
   const scenarioParameters = action.scenarioParameters;
-  const { error, data } = yield call(ScenarioService.updateAndLaunchScenario, ORGANISATION_ID, workspaceId, scenarioId, scenarioParameters);
-  if (error) {
-    // TODO handle error management
-    console.log(error);
+  // Update scenario parameters
+  const { error: updateError, data: updateData } = yield call(
+    ScenarioService.updateScenarioParameters, ORGANISATION_ID, workspaceId,
+    scenarioId, scenarioParameters);
+  if (updateError) {
+    console.error('Failed to update scenario parameters');
+    console.error(updateError);
+    yield put({
+      type: SCENARIO_ACTIONS_KEY.SET_CURRENT_SCENARIO,
+      data: { status: STATUSES.IDLE }
+    });
   } else {
-    console.log(data);
-    // Here is an effect named put that indicate to the middleware that it can dispatch a SET_CURRENT_SCENARIO action with list as payload
-    // TODO store data in new state
-    // yield put({ type: SCENARIO_ACTIONS_KEY.SET_CURRENT_SCENARIO, data: { status: STATUSES.SUCCESS, scenario: data } });
+    yield put({
+      type: SCENARIO_ACTIONS_KEY.SET_CURRENT_SCENARIO,
+      data: { status: STATUSES.SUCCESS, scenario: { parametersValues: updateData } }
+    });
+    // Launch scenario if parameters update succeeded
+    const { error: runError, data: runData } = yield call(
+      ScenariorunService.runScenario, ORGANISATION_ID, workspaceId, scenarioId);
+    if (runError) {
+      console.error(runError);
+    } else {
+      // Get csmSimulationRun id
+      const csmSimulationRun = runData.csmSimulationRun;
+      if (csmSimulationRun === undefined) {
+        console.error('csmSimulationRun is undefined');
+      } else {
+        yield put({
+          type: SCENARIO_ACTIONS_KEY.SET_CURRENT_SCENARIO,
+          data: {
+            status: STATUSES.SUCCESS,
```

Comment:
Should be RUNNING instead of SUCCESS.
At the moment, the simulation is running but is not necessarily finished.
The RUNNING status will allow us to display "Scenario running message" instead of ScenarioResultIframe, disable Scenario run button,... during a scenario run.
BUT, because we cannot know, for the moment, when a scenario launched, is finished, we need to trigger each X seconds (configurable in App.config.js) the scenario run status

Reply:
As discussed previously, this will be handled in PROD-7326 (branch THU/run_in_progress_PROD-7326)